packages/apm/src/span.ts

```diff
@@ -153,6 +153,19 @@ export class Span implements SpanInterface, SpanContext {
     return span;
   }
 
+  /**
+   * @inheritDoc
+   */
+  public getTransaction(): Transaction {
+    const recorder = this.spanRecorder;
+    if (!recorder) {
+      logger.warn('This Span has no reference to a Transaction. Returning an instance to itself.');
+      logger.warn('It means that the Transaction has been sampled or the Span did not originate from a Transaction.');
+      return (this as unknown) as Transaction;
```

Comment:
Since a span may not have an associated transaction in all cases, wouldn't it be clearer if this method was `getTransaction(): Transaction | undefined`?

Casting the current span into a transaction leaves downstream code no chance to handle this case, and all you get to work with is a warning if you have `debug: True`.